lib/matplotlib/cm.py

```diff
@@ -594,3 +633,35 @@ def changed(self):
         """
         self.callbacks.process('changed', self)
         self.stale = True
+
+
+# The docstrings here must be generic enough to apply to all relevant methods.
+mpl._docstring.interpd.update(
+    cmap_doc="""\
+cmap : str or `~matplotlib.colors.Colormap`, default: :rc:`image.cmap`
+    The Colormap instance or registered colormap name used to map scalar data
+    to colors. This parameter is ignored for RGB(A) data.""",
```

Comment:
I think not all functions using cmap support RGB(A) data. Now that we have entries per parameter it is possible to do move this out of the general description and do

```
%(cmap_doc)s
    
    This parameter is ignored for RGB(A) data.
```
only when needed (or make a `cmap_rgba_doc` if there are many cases).

Reply:
Actually, none of cmap, vmin/vmax, and norm apply for RGBA data, so all of them get this comment.

Reply:
Um, that's not what I meant. We have functions like [scatter](https://matplotlib.org/devdocs/api/_as_gen/matplotlib.axes.Axes.scatter.html) that allow either colormapped or RGBA data. Here, the sentence makes sense. But we also have functions like [hexbin](https://matplotlib.org/devdocs/api/_as_gen/matplotlib.axes.Axes.hexbin.html), that only operate with colormapping. Mentioning that a parameter is ignored for RGBA data when RGBA data is not supported by the function is somewhat confusing.

Reply:
Sorry I was unclear, I was meaning "all of cmap, norm, and vmin/vmax should get this comment".  From a quick check, the only functions that do not take color input are hexbin and hist2d; do you agree?

Reply:
From a quick search, I think the following functions support color input:

- [x] scatter
- [ ] hexbin
- [x] imshow
- [ ] pcolor
- [ ] pcolormesh
- [x] pcolorfast
- [ ] hist2d
- [ ] specgram
- [x] contour
- [x] tripcolor

Feel free to update the list.

Even if it's only hexbin and hist2d, how do we handle them?

Reply:
OK, this should be handled now.